index.js

```diff
@@ -448,6 +449,43 @@ module.exports = function(options) {
     return self.callAllAndEmit('afterInit', 'afterInit', callback);
   }
 
+  function lintModules(callback) {
+    _.each(self.modules, function(module, name) {
+      if (name.match(/-widgets$/) && (!extending(module)) && (!module.options.fromScratch)) {
```

Comment:
I'm not sold on `fromScratch`. Requiring an otherwise useless property to disable a warning the developer didn't ask for seems bad. I think these are useful warnings, but this element could use a few heads thinking. Maybe warnings can be disabled altogether? Maybe warnings should be opt-in?

As for the name, I'd recommend `noExtend` (I realize this comment is a variation on "The food was terrible, and such small portions.").

Reply:
Opt-in warnings would defeat the purpose. We need to grow our dev community, and clear communication about bugs they could waste a day on otherwise is key to that. It might annoy a few experienced devs, but probably not (because these are almost always real mistakes), and I'll take that over having people mystified during their early adoption phase and ending the process with "well this STINKS" after an hour or so of head-banging.

I did change the names of the options, you're right, those were no good:

`ignoreNoExtendWarning: true`
`ignoreNoCodeWarning: true`

Now we have a pattern that I think is sustainable, and we're not setting things to `false`, etc.

I do want to keep these simple, distinct options so we don't have to write a book in these messages about how to set them as an array or something.

Work for you?